lupa-core/src/main/kotlin/org/jetbrains/research/lupa/kotlinAnalysis/AnalysisExecutor.kt

```diff
@@ -32,23 +34,37 @@ abstract class AnalysisExecutor {
     /** Executes analysis for all projects in [given directory][projectsDir]. */
     fun execute(
         projectsDir: Path,
-        repositoryOpener: (Path, (Project) -> Unit) -> Unit = RepositoryOpenerUtil.Companion::standardRepositoryOpener
+        repositoryOpener: (Path, (Project) -> Unit, ((GitRepository) -> Unit)?) -> Unit =
```

Comment:
Also, should we change this function signature? I mean it is clear if the repositoryOpener receives a GitRepository

what do you think? In this case, we can run something like:
configData?.preExecuteAction(repo) (by default - an empty action)

I think it can be useful in the future